src/java.base/share/classes/java/lang/template/ValidatingProcessor.java

```diff
@@ -0,0 +1,163 @@
+ *
+ *              if (value instanceof Boolean) {
+ *                  throw new IllegalArgumentException("I don't like Booleans");
+ *              }
+ *
+ *              sb.append(value);
+ *          }
+ *
+ *          sb.append(fragmentsIter.next());
+ *
+ *          return sb.toString();
+ *     }
+ * }
+ *
+ * MyProcessor myProcessor = new MyProcessor();
+ * try {
+ *     int x = 10;
+ *     int y = 20;
+ *     String result = myProcessor."\{x} + \{y} = \{x + y}";
+ *     ...
+ * } catch (IllegalArgumentException ex) {
+ *     ...
+ * }
+ * }
+ * Implementations of this interface may provide, but are not limited to, validating
+ * inputs, composing inputs into a result, and transforming an intermediate string
+ * result to a non-string value before delivering the final result.
+ * <p>
+ * The user has the option of validating inputs used in composition. For example an SQL
+ * processor could prevent injection vulnerabilities by sanitizing inputs or throwing an
+ * exception of type {@code E} if an SQL statement is a potential vulnerability.
+ * <p>
+ * Composing allows user control over how the result is assembled. Most often, a
+ * user will construct a new string from the template string, with placeholders
+ * replaced by stringified objects from the values list.
+ * <p>
+ * Transforming allows the processor to return something other than a string. For
+ * instance, a JSON processor could return a JSON object, by parsing the string created
+ * by composition, instead of the composed string.
+ * <p>
+ * {@link ValidatingProcessor} is a {@link FunctionalInterface}. This permits
+ * declaration of a processor using lambda expressions;
+ * {@snippet :
+ * ValidatingProcessor<String, RuntimeException> processor = st -> {
+ *     List<String> fragments = st.fragments();
+ *     List<Object> values = st.values();
+ *     // check or manipulate the fragments and/or values
+ *     ...
+ *     return StringTemplate.interpolate(fragments, values);
+ * };
+ * }
+ * The {@link FunctionalInterface} {@link TemplateProcessor} is supplied to avoid
+ * the use of checked exceptions;
+ * {@snippet :
+ * TemplateProcessor<String> processor = st -> {
+ *     List<String> fragments = st.fragments();
+ *     List<Object> values = st.values();
+ *     // check or manipulate the fragments and/or values
+ *     ...
+ *     return StringTemplate.interpolate(fragments, values);
+ * };
+ * }
+ * The {@link FunctionalInterface} {@link StringProcessor} is supplied if
+ * the processor returns {@link String};
+ * {@snippet :
+ * StringProcessor processor = st -> {
+ *     List<String> fragments = st.fragments();
+ *     List<Object> values = st.values();
+ *     // check or manipulate the fragments and/or values
+ *     ...
+ *     return StringTemplate.interpolate(fragments, values);
+ * };
+ * }
+ * The {@link StringTemplate#interpolate()} method is available for those processors
+ * that just need to work with the interpolation;
```

Comment:
again, would be better to replace all bare occurrence of "interpolation" with "string interpolation"

Reply:
Changing.